From 5b747d74cf68998c0e6e8a7b54db1c6038857e4b Mon Sep 17 00:00:00 2001
From: jzhuan5 <jin.can.zhuang@intel.com>
Date: Fri, 29 Nov 2013 19:10:36 +0800
Subject: [PATCH 203/429] dwc3-device: fix fabric error caused by an unknown
 interrupt

After connect and disconnect A cable, then plug B cable, there's a
unknown interrupt received after requesting irq, and this interrupt will
be executed simultaneously w/ the core soft reset which messes up the
controller. Move the irq request after core soft reset to fix it.

Signed-off-by: jzhuan5 <jin.can.zhuang@intel.com>
---
 drivers/usb/dwc3/dwc3-device-intel.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/drivers/usb/dwc3/dwc3-device-intel.c b/drivers/usb/dwc3/dwc3-device-intel.c
index b4e6ad9989cd..dc53682edde6 100644
--- a/drivers/usb/dwc3/dwc3-device-intel.c
+++ b/drivers/usb/dwc3/dwc3-device-intel.c
@@ -187,12 +187,8 @@ int dwc3_start_peripheral(struct usb_gadget *g)
 	spin_unlock_irqrestore(&dwc->lock, flags);
 
 	irq = platform_get_irq(to_platform_device(dwc->dev), 0);
-	if (dwc->quirks_disable_irqthread)
-		ret = request_irq(irq, dwc3_quirks_interrupt,
-				IRQF_SHARED, "dwc3", dwc);
-	else
-		ret = request_threaded_irq(irq, dwc3_interrupt, dwc3_thread_interrupt,
-				IRQF_SHARED, "dwc3", dwc);
+	ret = request_threaded_irq(irq, dwc3_interrupt, dwc3_thread_interrupt,
+			IRQF_SHARED, "dwc3", dwc);
 	if (ret) {
 		dev_err(dwc->dev, "failed to request irq #%d --> %d\n",
 				irq, ret);
-- 
2.37.3

